buildSrc/subprojects/configuration/src/main/kotlin/org/gradle/gradlebuild/build-environment.kt

```diff
@@ -10,6 +10,8 @@ object BuildEnvironment {
     val jvm = org.gradle.internal.jvm.Jvm.current()
     val javaVersion = JavaVersion.current()
     val isWindows = OperatingSystem.current().isWindows
+    val isSlowInternetConnection
+        get() = System.getProperty("slow.internet.connection")?.toBoolean() ?: false
```

Comment:
I guess you could use `System.getProperty("slow.internet.connection", "false")` here.